microcosm_sagemaker/testing/pipeline_harness.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from inspect import signature
+
+
+class PipelineHarness(ABC):
+
+    """
+    Base class to support feature tests, by specifying the steps/pipes
+    of a pipeline
+    """
+
+    @property
+    @abstractmethod
+    def _steps(self):
```

Comment:
Would be good to use type annotation here.  I think probably
```suggestion
    def _steps(self) -> Iterable[Callable[..., Optional[Mapping]]]:
```
Docstring would be good too